.github/workflows/tests-on-pr.yml

```diff
@@ -13,7 +13,7 @@ jobs:
       run:
         shell: bash -l {0}
 
-    runs-on: ubuntu-latest
+    runs-on: macos-latest
```

Comment:
It turns out `wxpython` needs to be built for linux since no native whl is provided while we have windows/macos whl files provided shown below:

<img width="633" alt="Screenshot 2024-11-27 at 4 03 15 PM" src="https://github.com/user-attachments/assets/4dd7d163-e746-4b81-ba4b-51eb69aedb57">

Since @yucongalicechen and I test using macOS, what if we setup a MacOS CI instead?

Ref:
https://pypi.org/project/wxPython/#files

Reply:
Move back to Linux after we add wxpython to conda.txt